src/onefetch/info.rs

```diff
@@ -695,6 +792,61 @@ impl Info {
         };
         Some(color)
     }
+
+    fn get_formatted_subtitle_label(
+        &self,
+        label: &str,
+        color: Color,
+        colon_clr: Color,
+    ) -> ColoredString {
+        let formatted_label = format!("{}{} ", label.color(color), ":".color(colon_clr));
+        if self.config.no_bold {
+            formatted_label.normal()
+        } else {
+            formatted_label.bold()
+        }
+    }
+
+    fn get_color_set(text_colors: &[String], default_colors: &Vec<Color>) -> TextColor {
+        let mut custom_color: Vec<Color> = Vec::<Color>::new();
+        if text_colors.is_empty() {
+            custom_color.insert(0, default_colors[0]);
+            custom_color.insert(1, Color::White);
+            custom_color.insert(2, Color::White);
+            custom_color.insert(3, default_colors[0]);
+            custom_color.insert(4, default_colors[0]);
+            custom_color.insert(5, Color::White);
+        } else {
+            custom_color = text_colors
+                .iter()
+                .map(|color_num| {
+                    let custom = Info::num_to_color(color_num);
+                    match custom {
+                        Some(custom) => custom,
+                        None => Color::White,
+                    }
+                })
+                .collect();
+            if custom_color.len() < 6 {
+                for i in 0..6 {
+                    if i < custom_color.len() {
+                        continue;
+                    }
+                    custom_color.insert(i, Color::White);
+                }
+            }
+        }
+        
+        let color_set: TextColor = TextColor {
+            title: custom_color[0],
+            tilde: custom_color[1],
+            underline: custom_color[2],
+            subtitle: custom_color[3],
+            colon: custom_color[4],
+            info: custom_color[5],
+        };
+        color_set
+    }
```

Comment:
if you add a default value for TextColor:

```rust
impl Default for TextColor {
    fn default() -> TextColor {
        TextColor {
            title: Color::White,
            tilde: Color::White,
            underline: Color::White,
            subtitle: Color::White,
            colon: Color::White,
            info: Color::White,
        }
    }
}
```
This function can be a little bit simplified:

```rust
    fn get_text_color_set(text_colors: &[String], default_colors: &Vec<Color>) -> TextColor {
        let mut text_color_set = TextColor {
            ..Default::default()
        };

        if text_colors.is_empty() {
            text_color_set.title = default_colors[0];
            text_color_set.subtitle = default_colors[0];
            text_color_set.colon = default_colors[0];
        } else {
            let mut custom_color: Vec<Color> = Vec::<Color>::new();

            for i in 0..6 {
                if i < text_colors.len() {
                    let custom = match Info::num_to_color(&text_colors[i]) {
                        Some(custom) => custom,
                        None => Color::White,
                    };

                    custom_color.push(custom);
                } else {
                    custom_color.push(Color::White);
                }
            }

            text_color_set.title = custom_color[0];
            text_color_set.tilde = custom_color[1];
            text_color_set.underline = custom_color[2];
            text_color_set.subtitle = custom_color[3];
            text_color_set.colon = custom_color[4];
            text_color_set.info = custom_color[5];
        }

        text_color_set
    }
}
```